File: app/src/main/java/by/test/belarussian/belarussiantests/bizlogic/rvadapter/AdapterUtils.java
package by.test.belarussian.belarussiantests.bizlogic.rvadapter;

import android.support.v4.content.ContextCompat;
import android.support.v7.widget.CardView;
import android.view.View;
import android.widget.CheckBox;
import android.widget.ImageView;

import java.util.List;

import by.test.belarussian.belarussiantests.R;
import by.test.belarussian.belarussiantests.bizlogic.qmodel.Question;

public class AdapterUtils {
    void setAnswerImage(Question question, List<ImageView> imageViews) {
        for (int i = 0; i < imageViews.size(); i++) {
            if (question.getAnswers()[i].isCorrect()) {
                if (question.getAnswers()[i].isSelected()) {
                    imageViews.get(i).setImageResource(R.drawable.right);
                } else {
                    imageViews.get(i).setImageResource(R.drawable.wrong);
                }
            }
        }
    }

    void setDefaultAnswerImage(List<ImageView> imageViews) {
        for (ImageView imageView : imageViews) {
            imageView.setImageDrawable(null);
        }
    }

    void setCardViewColor(Question question, CardView cardView, View itemView) {
        if (question.isAnswered()) {
            cardView.setBackground(ContextCompat.getDrawable(itemView.getContext(),
                    R.drawable.border_green));
        } else {
            cardView.setBackground(ContextCompat.getDrawable(itemView.getContext(),
                    R.drawable.border_red));
        }
    }

    void setCheckBoxes(Question question, List<CheckBox> checkBoxes) {
        resetCheckBoxes(checkBoxes);
        for (int i = 0; i < question.getAnswers().length; i++) {
            checkBoxes.get(i).setClickable(false);
            if (question.getAnswers()[i].isSelected()) {
                checkBoxes.get(i).setChecked(true);
            }
        }
    }

    private void resetCheckBoxes(List<CheckBox> checkBoxes) {
        for (CheckBox checkBox : checkBoxes) {
            checkBox.setChecked(false);
        }
    }
}

File: app/src/main/java/by/test/belarussian/belarussiantests/bizlogic/utils/AdapterUtils.kt
@file:JvmName("AdapterUtils")

package by.test.belarussian.belarussiantests.bizlogic.utils

import android.support.v4.content.ContextCompat
import android.support.v7.widget.CardView
import android.view.View
import android.widget.CheckBox
import android.widget.ImageView
import by.test.belarussian.belarussiantests.R
import by.test.belarussian.belarussiantests.bizlogic.qmodel.Question

/**
 * File with utility methods for [RecycleViewAdapter]
 *
 * @author Pavel Davydzenka
 */
fun setAnswerImage(question: Question, imageViews: List<ImageView>) {
    for (i in imageViews.indices) {
        if (question.answers[i].isCorrect) {
            if (question.answers[i].isSelected) {
                imageViews[i].setImageResource(R.drawable.right)
            } else {
                imageViews[i].setImageResource(R.drawable.wrong)
            }
        }
    }
}

fun resetAnswerImage(imageViews: List<ImageView>) {
    for (imageView in imageViews) {
        imageView.setImageDrawable(null)
    }
}

fun setCardViewColor(question: Question, cardView: CardView, itemView: View) {
    if (question.isAnswered) cardView.background = ContextCompat.getDrawable(
            itemView.context,
            R.drawable.border_green)
    else cardView.background = ContextCompat.getDrawable(
            itemView.context,
            R.drawable.border_red)
}

fun setCheckBoxes(question: Question, checkBoxes: List<CheckBox>) {
    resetCheckBoxes(checkBoxes)
    for (i in 0 until question.answers.size) {
        checkBoxes[i].isClickable = false
        if (question.answers[i].isSelected) checkBoxes[i].isChecked = true
    }
}

private fun resetCheckBoxes(checkBoxes: List<CheckBox>) {
    for (checkBox in checkBoxes) {
        checkBox.isChecked = false
    }
}